src/main/kotlin/com/icat/orboarding/user/adapters/configuration/TokenAuthFilterConfiguration.kt

```diff
@@ -0,0 +1,54 @@
+package com.icat.orboarding.user.adapters.configuration
+
+import com.icat.orboarding.user.adapters.outbound.persistence.mysql.entities.UserEntity
+import com.icat.orboarding.user.adapters.outbound.persistence.mysql.repositories.UserRepository
+import com.icat.orboarding.user.application.services.TokenService
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.stereotype.Component
+import org.springframework.web.filter.OncePerRequestFilter
+import java.io.IOException
+import java.util.*
+import javax.servlet.FilterChain
+import javax.servlet.ServletException
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+
+@Component
+class TokenAuthFilterConfiguration(
+    private val tokenService: TokenService,
+    private val repository: UserRepository
+    ) : OncePerRequestFilter() {
+
+    @Throws(ServletException::class, IOException::class)
+    override fun doFilterInternal(
+        request: HttpServletRequest,
+        response: HttpServletResponse,
+        filterChain: FilterChain
+    ) {
+        val tokenFromHeader: String? = getTokenFromHeader(request)
```

Comment:
```suggestion
        val tokenFromHeader = getTokenFromHeader(request)
```
Como o método `getTokenFromHeader` retorna uma String, aqui você pode passar diretamente ser deixar a tipagem explicita